poppy/wavefront.py

```diff
@@ -0,0 +1,359 @@
+from __future__ import division
+import numpy as np
+import matplotlib.pyplot as plt
+import scipy.constants as const
+import astropy.units as u
+from copy import deepcopy
+from scipy.linalg import lstsq
+
+from poppy.fresnel import FresnelWavefront, QuadraticLens
+
+
+class PhysicalFresnelWavefront(FresnelWavefront):
+    """
+    This class extends the capabilities of poppy's FresnelWavefront class.
+    
+    The extension includes the redefinition of the wavefront array to
+    represent an electric field with the appropriate units V.m^-1.
+    Some utility functions have been defined to be able to compute the beam
+    radius and beam quality factor. Additionally, a wavelength scaling has 
+    been introduced to simulate a beam quality factor.
+    
+    Parameters
+    ----------
+    M2 : float
+        Beam quality factor.
+    n0 : float
+        Ambient refractive index.
+    """
+    
+    def __init__(self,
+                 beam_radius,
+                 units=u.m,
+                 rayleigh_factor=2.0,
+                 oversample=2,
+                 wavelength=1064.0e-9,
```

Comment:
Generally in the other functions I've tried to leave the defaults at round numbers rather than baking in particular science/engineering assumptions about wavelengths.  So I'm a hesitant about having this default to Nd:YAG.  I also don't like having the different Wavefront classes have different defaults for a common parameter; that's not the least surprising interface. But hmm, I just realized the defaults on the other wavefronts are 2 microns, which relates back to the main science wavelengths for NASA's JWST mission, so that's not a great default either.   Now I think I might want to make the default 1 micron for all 3 of the Wavefront, FresnelWavefront, and PhysicalFresnelWavefront classes, so they would be consistent.